tests/asgi/tests.py

```diff
@@ -376,6 +378,14 @@ async def test_concurrent_async_uses_multiple_thread_pools(self):
     async def test_asyncio_cancel_error(self):
         # Flag to check if the view was cancelled.
         view_did_cancel = False
+        finished_threads = []
+
+        @async_unsafe
+        def on_finished(sender, **kwargs):
+            nonlocal finished_threads
+            finished_threads.append(threading.current_thread())
+
+        request_finished.connect(on_finished)
```

Comment:
I'm wondering the reason for choosing this approach for recording the `request_finished` thread instead of following the existing code that uses the `SignalHandler`. I know the latter is a bit more... verbose but it would avoid the `nonlocal` call and it will also be more consistent/less surprising when looked in conjunction with the rest of the tests.

Reply:
Ok sure I can update this to clone the style use in the other tests